test/src/unit-capi-array.cc

```diff
@@ -2102,25 +2102,50 @@ TEST_CASE_METHOD(
   rc = tiledb_array_open(ctx_, new_array, TILEDB_READ);
   REQUIRE(rc == TILEDB_OK);
 
-  int is_empty;
-  uint64_t domain[4];
-  rc = tiledb_array_get_non_empty_domain(ctx_, new_array, domain, &is_empty);
+  // Check the retrieved array schema
+  tiledb_array_schema_t* new_array_schema;
+  rc = tiledb_array_schema_load(ctx_, array_name.c_str(), &new_array_schema);
   CHECK(rc == TILEDB_OK);
-  CHECK(is_empty == 1);
 
-  // Close new_array
-  rc = tiledb_array_close(ctx_, new_array);
-  CHECK(rc == TILEDB_OK);
-  tiledb_array_free(&new_array);
+  rc = tiledb_array_schema_check(ctx_, new_array_schema);
+  REQUIRE(rc == TILEDB_OK);
 
-  /** Validate metadata. **/
-  // Open new_array in read mode
-  rc = tiledb_array_alloc(ctx_, array_name.c_str(), &new_array);
+  tiledb_layout_t cell_order;
+  rc = tiledb_array_schema_get_cell_order(ctx_, new_array_schema, &cell_order);
   REQUIRE(rc == TILEDB_OK);
-  rc = tiledb_array_open(ctx_, new_array, TILEDB_READ);
+  CHECK(cell_order == TILEDB_ROW_MAJOR);
+
+  tiledb_layout_t tile_order;
+  rc = tiledb_array_schema_get_tile_order(ctx_, new_array_schema, &tile_order);
+  REQUIRE(rc == TILEDB_OK);
+  CHECK(tile_order == TILEDB_ROW_MAJOR);
+
+  unsigned int num_attributes = 0;
+  rc = tiledb_array_schema_get_attribute_num(
+      ctx_, new_array_schema, &num_attributes);
+  REQUIRE(rc == TILEDB_OK);
+  CHECK(num_attributes == 1);
+
+  tiledb_domain_t* dom;
+  rc = tiledb_array_schema_get_domain(ctx_, new_array_schema, &dom);
   REQUIRE(rc == TILEDB_OK);
 
-  // Read
+  unsigned int ndim = 0;
+  rc = tiledb_domain_get_ndim(ctx_, dom, &ndim);
+  REQUIRE(rc == TILEDB_OK);
+  CHECK(ndim == 2);
+
+  // TODO: Check the retrieved array_schemas list. Currently there is no
+  // CAPI to get it, should we introduce one?
```

Comment:
I would first double check if we can get the list from the underlying sm pointer.

Reply:
Done, thanks for the suggestion!